aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -195,6 +195,59 @@ Tensor pinverse(const Tensor& self, double rcond) {
   return at::linalg_pinv(self, rcond, /*hermitian=*/false);
 }
 
+Tensor& linalg_matrix_power_out(const Tensor& self, int64_t n, Tensor& result) {
+  checkSameDevice("matrix_power", result, self);
+  checkLinalgCompatibleDtype("matrix_power", result, self);
+  at::native::resize_output(result, self.sizes());
+  result.copy_(at::native::linalg_matrix_power(self, n));
+  return result;
+}
+
+Tensor linalg_matrix_power(const Tensor& self, int64_t n) {
+  squareCheckInputs(self);
+
+  TORCH_CHECK(
+      at::isFloatingType(self.scalar_type()) ||
+          at::isComplexType(self.scalar_type()),
+      "matrix_power: expected input tensor of floating or complex type but got ",
+      self.scalar_type());
+
+  auto a = n < 0 ? self.inverse() : self.clone(at::MemoryFormat::Contiguous);
```

Comment:
Let's start using new `at::linalg_inv` instead of `at::inverse/.inverse` for the internal code as well.